.changeset/bright-seas-march.md

```diff
@@ -0,0 +1,11 @@
+---
+'@leafygreen-ui/confirmation-modal': minor
+---
+
+- Drops the `isRequired` from the `buttonText` prop type. As of [version 5.1.0](https://github.com/mongodb/leafygreen-ui/blob/main/packages/confirmation-modal/CHANGELOG.md#510), `buttonText` is a deprecated prop that is now optional.
+- Fixes existing confirm button disabled state logic
+  - Previously, when the modal was closed using the cancel or confirm button and reopened, the confirm button would always reset to a disabled state although it should not reset when `requiredInputText` is provided.
+  - Now, the confirm button will reset to a disabled state when `requiredInputText` is provided.
```

Comment:
Do these two bullets contradict each other? 

> the confirm button would always reset to a disabled state although it should not reset when `requiredInputText` is provided

and 

> Now, the confirm button will reset to a disabled state when `requiredInputText` is provided.

Reply:
good catch, updated! I think it should read better now 😅